contents/java/Spring.md

```diff
@@ -0,0 +1,277 @@
+</tooltip>in Spring Framework.
+
+In the Spring framework, the IoC container is represented by the interface `ApplicationContext`.
+The beans are created with the configuration metadata that you supply to the container,
+which can be in the form of XML configuration or annotations.
+
+Below is one way to manually instantiate a container:
+```java
+ApplicationContext context = new ClassPathXmlApplicationContext("applicationContext.xml");
+```
+
+For example, say we have an Employee class like so:
+```java
+package com.company;  
+  
+public class Employee {  
+    private int id;  
+    private String name;  
+    private String city;  
+  
+    public int getId() {  
+        return id;  
+    }  
+    public void setId(int id) {  
+        this.id = id;  
+    }  
+    public String getName() {  
+        return name;  
+    }  
+    public void setName(String name) {  
+        this.name = name;  
+    }  
+    void display(){  
+        System.out.println(id+" "+name);  
+    }
+}        
+```
+
+We can then populate the Employee bean with the data provided in the following `applicationContext.xml` file:
+This makes our code loosely coupled and easier for testing. 
+```xml
+<bean id="employeeAmy" class="com.company.Employee">  
+    <property name="id">  
+        <value>20</value>  
+    </property>  
+    <property name="name">  
+        <value>Amy</value>  
+    </property>  
+</bean>  
+```
+   
+To better understand how this dependency injection works, we can have a `Test.java` looks like the following:
+```java
+Resource r = new ClassPathResource("applicationContext.xml");  
+BeanFactory factory = new XmlBeanFactory(r);  
+          
+Employee e = (Employee)factory.getBean("obj");  
+e.display();
+```
+
+The output of the above code would then be `20 Amy`.
+
+### Spring JdbcTemplate
+
+The Spring JdbcTemplate is a powerful mechanism to connect to the database and execute SQL queries. Traditionally, we
+need to use Java Database Connectivity (JDBC) API to access tabular data stored in any relational database.
+It is a part of JavaSE (Java Standard Edition), from Oracle Corporation.
+    
+In a traditional JDBC API:
+- a typical SQL query would involve quite some amount of boilerplate code, such as creating connection,
+statement, closing resultset, connection etc.
+- exception handling has to be manually written out.
+- transactions need to be manually created.
+
+The Spring JdbcTemplate provides methods to write the queries directly, hence, saving time in writing such boilerplate code.
+
+Task | Spring | You
+-----|--------|----
+Connection Management | :far-check-circle: | 
+Writing SQL Queries | | :far-check-circle: 
+Statement Management | :far-check-circle: |
+ResultSet Management | :far-check-circle: |
+Row Data Retrieval | | :far-check-circle: 
+Parameter Declaration | | :far-check-circle: 
+Parameter Setting | :far-check-circle: |
+Transaction Management | :far-check-circle: |
+Exception Handling | :far-check-circle: |
+
```

Comment:
It's nice if you can give example code for both, for a comparison. OK if you can't find such code easily.